src/Servers/Kestrel/Transport.Sockets/src/Internal/IOQueue.cs

```diff
@@ -73,4 +75,19 @@ public Work(Action<object?> callback, object? state)
             State = state;
         }
     }
+
+    private static int DetermineDefaultCount()
+    {
+        // Since each IOQueue schedules one work item to process its work, the number of IOQueues determines the maximum
+        // parallelism of processing work queued to IOQueues. The default number below is based on the processor count and tries
+        // to use a high-enough number for that to not be a significant limiting factor for throughput.
+
+        int processorCount = Environment.ProcessorCount;
+        if (OperatingSystem.IsWindows() || processorCount <= 32)
```

Comment:
Out of curiosity: why is Windows-only?

Reply:
Forgot to mention in the comment, will update. On Windows due to other perf issues at the moment using more IOQueues is hurting rather than helping. Hoping that would be fixed at some point, at which time the same change would apply to Windows as well.